src/fides/api/models/sql_models.py

```diff
@@ -514,6 +532,124 @@ def create(
         """Overrides base create to avoid unique check on `name` column"""
         return super().create(db=db, data=data, check_name=check_name)
 
+    @hybrid_property
+    def purpose(self) -> Optional[int]:
+        """Returns the instance-level TCF Purpose."""
+        mapped_purpose: Optional[MappedPurpose] = MAPPED_PURPOSES_ONLY_BY_DATA_USE.get(
+            self.data_use
+        )
+        return mapped_purpose.id if mapped_purpose else None
+
+    @purpose.expression
+    def purpose(cls) -> Case:
+        """Returns the class-level TCF Purpose"""
+        return case(
+            [
+                (cls.data_use == data_use, purpose.id)
+                for data_use, purpose in MAPPED_PURPOSES_ONLY_BY_DATA_USE.items()
+            ],
+            else_=None,
+        )
+
+    @hybrid_property
+    async def _publisher_override_legal_basis_join(self) -> Optional[str]:
+        """Returns the instance-level overridden required legal basis"""
+        query: Select = select([TCFPublisherOverride.required_legal_basis]).where(
+            TCFPublisherOverride.purpose == self.purpose
+        )
+        async_session: AsyncSession = async_object_session(self)
+        async with async_session.begin():
+            result = await async_session.execute(query)
+            return result.scalars().first()
+
+    @_publisher_override_legal_basis_join.expression
+    def _publisher_override_legal_basis_join(cls) -> ScalarSelect:
+        """Returns the class-level overridden required legal basis"""
+        return (
+            select([TCFPublisherOverride.required_legal_basis])
+            .where(TCFPublisherOverride.purpose == cls.purpose)
+            .as_scalar()
+        )
+
+    @hybrid_property
+    async def _publisher_override_is_included_join(self) -> Optional[bool]:
+        """Returns the instance-level indication of whether the purpose should be included"""
+        query: Select = select([TCFPublisherOverride.is_included]).where(
+            TCFPublisherOverride.purpose == self.purpose
+        )
+        async_session: AsyncSession = async_object_session(self)
+        async with async_session.begin():
+            result = await async_session.execute(query)
+            return result.scalars().first()
+
+    @_publisher_override_is_included_join.expression
+    def _publisher_override_is_included_join(cls) -> ScalarSelect:
+        """Returns the class-level indication of whether the purpose should be included"""
```

Comment:
These are supporting instance-level and class-level hybrid properties for the primary overridden_legal_basis_for_processing properties below